network/p2p/unicast/ratelimit/internal/limiter_map/rate_limiter_map.go

```diff
@@ -0,0 +1,109 @@
+package limiter_map
+
+import (
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+type RateLimiterMetadata struct {
+	// Limiter the rate Limiter
+	Limiter *rate.Limiter
+	// LastRateLimit the last timestamp this the peer was rate limited.
+	LastRateLimit time.Time
```

Comment:
Given my previous comment about race conditions these _shouldn't_ be exported
```suggestion
	// lastRateLimit the last timestamp this the peer was rate limited.
	lastRateLimit time.Time
```

Reply:
https://github.com/onflow/flow-go/pull/2840/commits/1702077fca9c875c8f564403a6ebad3a99afa923 Adds mutex and un-exports those fields.